.clang-tidy

```diff
@@ -0,0 +1,81 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+# Modified from: https://raw.githubusercontent.com/googleapis/google-cloud-cpp/main/.clang-tidy
+# See: https://releases.llvm.org/10.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/list.html
+---
+# Configure clang-tidy for this project.
+
+# Here is an explanation for why some of the checks are disabled:
+#
+#  -modernize-use-trailing-return-type: clang-tidy recommends using
+#      `auto Foo() -> std::string { return ...; }`, we think the code is less
+#      readable in this form.
+#
+#  -modernize-return-braced-init-list: We think removing typenames and using
+#      only braced-init can hurt readability.
+#
+#  -modernize-avoid-c-arrays: We only use C arrays when they seem to be the
+#      right tool for the job, such as `char foo[] = "hello"`. In these cases,
+#      avoiding C arrays often makes the code less readable, and std::array is
+#      not a drop-in replacement because it doesn't deduce the size.
+#
+#  -google-readability-braces-around-statements: This is a duplicate of
+#      readability-braces-around-statements, which is configured below
+#      to allow single-line statements without braces.
+#
+#  -google-runtime-references: Allow usage of non-const references as
+#      function parameters. Otherwise we'd have to use pointers, which
+#      cpp core guidelines recommends against unless the parameter is
+#      nullable: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f60-prefer-t-over-t-when-no-argument-is-a-valid-option
+#
+#
+Checks: >
+    -*,
+    bugprone-*,
+    google-*,
+    misc-*,
+    modernize-*,
+    performance-*,
+    portability-*,
+    readability-*,
+    -google-readability-braces-around-statements,
+    -google-runtime-references,
+    -modernize-return-braced-init-list,
+    -modernize-use-trailing-return-type,
+    -modernize-avoid-c-arrays,
+
+# Turn all the warnings from the checks above into errors.
+WarningsAsErrors: "*"
+# Scan all (non-system) headers.
+HeaderFilterRegex: '.*'
+# Use .clang-format for fix suggestions.
+FormatStyle: file
+
+CheckOptions:
+    - { key: readability-identifier-naming.NamespaceCase,          value: camelBack  }
+    - { key: readability-identifier-naming.ClassCase,              value: CamelCase  }
+    - { key: readability-identifier-naming.StructCase,             value: CamelCase  }
+    - { key: readability-identifier-naming.TemplateParameterCase,  value: CamelCase  }
+    - { key: readability-identifier-naming.FunctionCase,           value: camelBack  }
+    - { key: readability-identifier-naming.VariableCase,           value: camelBack  }
+    - { key: readability-identifier-naming.ClassMemberCase,        value: camelBack  }
+    - { key: readability-identifier-naming.ClassMemberSuffix,      value: _          }
```

Comment:
Not sure about this if it includes the public API.

Reply:
Updated in fca16de4